### ISSUE TITLE
Can DISAM be applied in federated learning scenarios？

### ISSUE THREAD

**Issue by xxdznl** [MediaBrain-SJTU/DISAM#2]
Thanks for the splendid work, I am very admirable。

1. The different domains mentioned in the paper are similar to the **non-IID problem** scenarios in **federated learning**. Is this an application direction of DISAM? 
2. In addition, can DISAM be used in one-domain datasets such as MNIST, CIFAR, and ImageNet?

Sorry for my ignorance in the field of domain generalization. Thank you very much for taking the time to address my confusion.

**Comment by FrankZhangRp** [comment]
[2024.6.19 update] Our team's new work on federated learning with SAM is released here: https://github.com/MediaBrain-SJTU/FedLESAM

---

Thank you for paying attention to our work.

1. In fact, SAM-based methods have already been widely used in Federated Learning, such as FedSAM, FedASAM, and FedGEMMA. However, our DISAM requires knowing the domain labels of multiple domains during training. If the domains of different clients are different, DISAM cannot be directly used. Our team has recently submitted a new SAM-based FL method to ICML 2024, which will be made public soon. You may want to follow it.

2. DISAM aims to address the issue of multi-source training data. When the training data has only one distribution, there is no issue of inconsistent convergence between different distributions, and DISAM will degenerate to SAM. Alternatively, some clustering methods can be used to discover multiple latent domains, and DISAM can be trained using the latent domain labels.
